projects/igniteui-angular/src/lib/tree/tree-node/tree-node.component.ts

```diff
@@ -0,0 +1,355 @@
+    providers: [
+        { provide: IGX_TREE_NODE_COMPONENT, useExisting: IgxTreeNodeComponent }
+    ]
+})
+export class IgxTreeNodeComponent<T> extends ToggleAnimationPlayer implements IgxTreeNode<T>, OnInit, AfterViewInit, OnDestroy {
+    @Input()
+    public data: T;
+
+    public set animationSettings(val: ToggleAnimationSettings) { }
+
+    public get animationSettings(): ToggleAnimationSettings {
+        return this.tree.animationSettings;
+    }
+
+    /** @hidden @internal */
+    private _resourceStrings = CurrentResourceStrings.TreeResStrings;
+
+    /**
+     * An accessor that sets the resource strings.
+     * By default it uses EN resources.
+     */
+    @Input()
+    public set resourceStrings(value: ITreeResourceStrings) {
+        this._resourceStrings = Object.assign({}, this._resourceStrings, value);
+    }
+
+    /**
+     * An accessor that returns the resource strings.
+     */
+    public get resourceStrings(): ITreeResourceStrings {
+        if (!this._resourceStrings) {
+            this._resourceStrings = CurrentResourceStrings.TreeResStrings;
+        }
+        return this._resourceStrings;
+    }
+
+    /**
+     * Emitted when the node's `selected` property changes.
+     *
+     * ```html
+     * <igx-tree>
+     *      <igx-tree-node *ngFor="let node of data" [data]="node" [(selected)]="data.selected">
+     * </igx-tree>
+     * ```
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[0])[0];
+     * node.selectedChange.pipe(takeUntil(this.destroy$)).subscribe((e: boolean) => console.log("Node selection changed to ", e))
+     * ```
+     */
+    @Output()
+    public selectedChange = new EventEmitter<boolean>();
+
+    // TODO: bind to active state when keynav is implemented
+    /** @hidden @internal */
+    @HostBinding('class.igx-tree-node--active')
+    public get active() {
+        return false;
+    }
+
+    // TODO: bind to disabled state when node is dragged
+    /** @hidden @internal */
+    @HostBinding('class.igx-tree-node--disabled')
+    public get disabled() {
+        return false;
+    }
+
+    /** @hidden @internal */
+    @HostBinding('class.igx-tree-node')
+    public cssClass = 'igx-tree-node';
+
+    /** @hidden @internal */
+    @HostBinding('attr.role')
+    public roleAttr = 'treeitem';
+
+    // TODO: Public API should expose array or null, not query list
+    @ContentChildren(IGX_TREE_NODE_COMPONENT, { read: IGX_TREE_NODE_COMPONENT })
+    public children: QueryList<IgxTreeNode<any>>;
+
+    // TODO: Expose in public API instead of `children` query list
+    /**
+     * Return the child nodes of the node (if any)
+     *
+     * @remark
+     * Returns `null` if node does not have children
+     *
+     * ```typescript
+     * const node: IgxTreeNode<any> = this.tree.findNodes(data[0])[0];
+     * const children: IgxTreeNode<any>[] = node.children;
+     * ```
+     */
+    public get _children(): IgxTreeNode<any>[] {
```

Comment:
Should the public getter be with `_`?

Reply:
This will be change once all functionality is implemented - we'll remove the `_` and swap this w/ the current `children: QueryList<IgxTreeNode<any>>` property. I've left a `TODO` comment above the API docs for this one.
There's currently a `children: QueryList<IgxTreeNode>` property, which is `/** @hidden @internal */` and is used in the logic for selection and navigation.